src/Microsoft.AspNet.Server.Kestrel/Http/FrameHeaders.Generated.cs

```diff
@@ -6,10 +6,10 @@ namespace Microsoft.AspNet.Server.Kestrel.Http
 {
     public partial class FrameResponseHeaders
     {
-        public FrameResponseHeaders()
+        public FrameResponseHeaders(string dateHeaderValue)
         {
             _Server = new[] { "Kestrel" };
-            _Date = new[] { DateTime.UtcNow.ToString("r") };
+            _Date = new[] { dateHeaderValue };
```

Comment:
This needs to be set at the end of the response rather than the beginning. Things like the error handling middleware will clear the response headers.